projects/components/facet/bootstrap/facet-filters/facet-filters.ts

```diff
@@ -130,4 +138,66 @@ export class BsFacetFilters implements OnChanges {
         return this.facetService.hasData(facet.aggregation, this.results);
     }
 
+    private addFacetMenu() {
+        if (!this.userFacets || this.userFacets?.length < this.facets.length) { 
+            let outFacets: Action[] = [];
+
+            outFacets.push(new Action({
+                name: `add_all`,
+                text: "All",
+                title: `Add all facets`,
+                action: () => {
+                    if (this.userSettingsService.userSettings) this.userSettingsService.userSettings["facets"] = undefined;
+                    this.userSettingsService.save();
+                    this.initUserFacets();
+                    this.buildFilters();
+                }
+            }));
+
+            for (let facet of this.facets) {
```

Comment:
Another approach could be to use `action.selected` (which displays a checkbox). That way you have one action for turning on or off each facet. The legacy `enableAddFacet` is something that we should probably remove.

Reply:
Not sure what you mean about this part... The where should the action.selected be used and how can I add a checkbox as an action of the facet, given that the actions are all defined in the original BsFacetList and BsFacetTree components? Or should I add it at the bottom just like it is now with the enableAddFacet?

Reply:
Right now you create an action for each facet to add this facet. But instead this action could a checkbox (when checked the facet is added, when unchecked it is removed). To do that, it looks like:

```
new Action({
                text: "Toto",
                selected: true,
                action: action => {
                    action.selected = !action.selected; // toggle checkbox
                    if(action.selected) this.addFacet(...)
                    else this.removeFacet(...);
                }
            })

```